persistent-postgresql/Database/Persist/Postgresql.hs

```diff
@@ -1738,6 +1754,176 @@ repsertManySql ent n = putManySql' conflictColumns fields ent n
     fields = keyAndEntityFields ent
     conflictColumns = escapeF . fieldDB <$> entityKeyFields ent
 
+-- | This type is used to determine how to update rows using Postgres'
+-- @INSERT ... ON CONFLICT KEY UPDATE@ functionality, exposed via
+-- 'upsertWhere' and 'upsertManyWhere' in this library.
+--
+-- @since 2.12.1
+data HandleUpdateCollision record where
+  -- | Copy the field directly from the record.
+  CopyField :: EntityField record typ -> HandleUpdateCollision record
+  -- | Only copy the field if it is not equal to the provided value.
+  CopyUnlessEq :: PersistField typ => EntityField record typ -> typ -> HandleUpdateCollision record
+
+-- | An alias for 'HandleUpdateCollision'. The type previously was only
+-- used to copy a single value, but was expanded to be handle more complex
+-- queries.
+--
+-- @since 2.12.1
+type SomeField = HandleUpdateCollision
```

Comment:
the rest of these types (L1773-1812) I copied over from `MySQL.hs`.  Do they make sense here?  It made everything compile but I'm not really sure what the difference between copying vs importing was.

Reply:
Copying means that they are *distinct* - the type `Database.Persist.Postgresql.HandleUpdateCollision` is not the same as the one defined in `Database.Persist.MySQL`, and you'll get a type error if you try to conflate the two.

I'd probably *not* copy this type synonym - it's a backwards compatibility thing. Since we're starting fresh, we can omit it.

Reply:
good call, and thanks for the explanation!